### ISSUE TITLE
I am curious if you have benchmarked this against octree approaches

### ISSUE THREAD

**Issue by RallyTronics** [Jaybro/pico_tree#8]
such as [Efficient Radius Neighbor Search in Three-dimensional Point Clouds](https://github.com/jbehley/octree)

**Comment by Jaybro** [comment]
Short answer: No. 

I wrote this KdTree for fun so that it could replace nanoflann in another project of mine. Nanoflann is known to be pretty fast, so I wanted to see if I could make a faster one and created the benchmark you see on this repository. That was good enough for me and it also takes quite some time to do a benchmark :)

Your question did get me a bit curious but I couldn't easily find any good comparisons between the structures. It also doesn't help that depending on the quality or goal of the implementation the performance can vary a lot as well. As an example on the goal, I wrote mine specifically for nearest neighbor queries at the price of having slower range/box searches by making certain implementation choices. 

I'd say that both structures have their ups and downs but this is a bit more theoretical:
* Depending on the distribution of points the depth of an octree is variable. A median split kdtree results in balanced tree (but the midpoint split is more similar to an octree).
* Mutation of kdtrees is more difficult than on octrees.
* Etc.
